install-deps.sh

```diff
@@ -146,9 +146,29 @@ function install_pkg_on_ubuntu {
     fi
 }
 
+boost_ver=1.82
+
+function clean_boost_on_ubuntu {
```

Comment:
It looks like `install_boost_on_ubuntu()` already has some code for removing undesired packages.  Can you summarize why it's not working in this case?

Did you consider expanding it instead of duplicating some of it in a new function?

Reply:
Two reasons. First, it wasn't being called because in Quincy we didn't have a case for Jammy. Since under the proposed Quincy fix Jammy doesn't call `install_boost_on_ubuntu()` (as it does not use the ceph-libboost packages), we need a separate clean function.

Second, the file it removes from `/etc/apt/sources.list.d` is not actually the file `install_boost_on_ubuntu()` installs there.

I just repushed to remove the relevant stuff from `install_boost_on_ubuntu()` though, so it's not duplicated.

Reply:
the commit message reads

> To force removal of overly new libboost packages when building with system boost.

but i see there are couple things in this change


1. extract "clean_boost_on_ubuntu()" out of "install_boost_on_ubuntu()" // you explained this in the comment above. but it's not explained in the commit message.
2. always call "install_pkg_on_ubuntu()" in "install_boost_on_ubuntu()" even if some boost package(s) is found with the expected version
3. always call "clean_boost_on_ubuntu()"
4. remove more source lists in `clean_boost_on_ubuntu()` // i can guess, but i don't really understand
5. list the source list  // for debugging?
6. call `apt-get update` in `clean_boost_on_ubuntu()` // what is it for?
7. remove all liboost `xargs` // i guess it's an improvement, but it deserves some explanation as well

in short, the change is dense. and i failed to read between the lines. could you please put more details on why and why and why in the commit message?

Reply:
It got more complicated pulling it onto main. Really I think it should have just been done against Quincy and that's it because nothing after Quincy ever had the problem of there NOT being a call to `install_boost_on_ubuntu()` for Jammy

Reply:
> Really I think it should have just been done against Quincy and that's it because nothing after Quincy ever had the problem of there NOT being a call to `install_boost_on_ubuntu()` for Jammy

`make check` on pacific is affected the same way.  I don't know if the root cause is the same, but it's not a quincy-only issue.

Also, is there anything to guarantee that this problem would not repeat with a different combination of Ceph and Ubuntu LTS releases in the future?

Reply:
I believe a straightforward backport of both commits needed for Quincy would fix Pacific as well.

Putting this on main wouldn't stop problems in stable releases if we switch LTS's either. It would mean they have the wrong boost removed, but nothing installs the right boost, because there's no case for the new LTS in the old install-deps.sh, and no version of an old boost package built against it.

At present it looks like if we *just* remove the wrong boost from Quincy and push then jenkins will fetch the boost source and build it, which I honestly didn't expect to work and I wouldn't really count on working in the future.

Reply:
>     2. always call "install_pkg_on_ubuntu()" in "install_boost_on_ubuntu()" even if some boost package(s) is found with the expected version

This was a mistake, thank you.

>     5. list the source list  // for debugging?

For debugging, yes, but I didn't mean to leave it in. Thank you.

Reply:
> At present it looks like if we _just_ remove the wrong boost from Quincy and push then jenkins will fetch the boost source and build it, which I honestly didn't expect to work and I wouldn't really count on working in the future.

I think this is expected and actually _can_ be counted upon.  At a high level:
- we try system boost
- if system boost isn't good enough, we try "ceph" pre-built boost
- if "ceph" pre-built boost isn't there/good enough, we download and build from scratch

When the build that you produced for 1.82 got pruned (I think that is what happened?), `make check` continued working precisely because of that.

Reply:
In general, yes.

But since more and more people are locking down their builders' ability to access the Internet, I don't expect it to keep working indefinitely.

But I don't care, I'll do either one. I just want to not leave quincy and pacific builders broken longer than necessary.